relayer/pkg/chainlink/ocr2/client.go

```diff
@@ -171,10 +168,10 @@ func (c *Client) LinkAvailableForPayment(ctx context.Context, address *felt.Felt
 		Selector:        starknetutils.GetSelectorFromNameFelt("link_available_for_payment"),
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to call the contract with selector 'link_available_for_payment'")
+		return nil, fmt.Errorf("failed to call the contract with selector 'link_available_for_payment': %w", err)
 	}
-	if len(results) != 2 {
-		return nil, errors.Wrap(err, "insufficient data from selector 'link_available_for_payment'")
+	if l := len(results); l != 2 {
+		return nil, fmt.Errorf("insufficient data from selector 'link_available_for_payment': need 2 results but got %d", l)
```

Comment:
This was accidentally wrapping a nil error, and so always returning nil. Now that this failure path is fixed, `TestOCR2Client/get_link_available_for_payment` is failing.